src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -404,14 +405,43 @@ static String validateConfig(Map<String, String> config)
     return connectorName;
   }
 
+  /**
+   * modify invalid application name in config and return the generated application name
+   * @param config input config object
+   */
+  public static void convertAppName(Map<String, String> config)
+  {
+    String appName =
+      config.getOrDefault(SnowflakeSinkConnectorConfig.NAME, "");
+    boolean appendHash = Boolean.parseBoolean(config.getOrDefault(
+      SnowflakeSinkConnectorConfig.APPEND_TABLE_HASH,
+      String.valueOf(SnowflakeSinkConnectorConfig.APPEND_TABLE_HASH_DEFAULT)));
```

Comment:
Use `snowflake.table.appendhash` flag to disable hash in connector name too. Connector name is used as prefix in Snowflake stage name.